### PR TITLE
refactor: use netip.ParsePrefix replace the net.ParseCIDR

### PR DESCRIPTION
**Pull request by saltbo**
**Please provide a description of this PR:**
try to partially implement issue #39348

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [x] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [x] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by saltbo** [comment]
test field for 192.168/16, is it a valid CIDR?

**Comment by ramaraochavali** [comment]
> test field for 192.168/16, is it a valid CIDR?

If it worked previously, it should be. Is the new API introducing stricter validation?

**Comment by saltbo** [comment]
> > test field for 192.168/16, is it a valid CIDR?
> 
> If it worked previously, it should be. Is the new API introducing stricter validation?

I think the previous code is a non-standard implementation.

And the old API is always not supported, https://go.dev/play/p/cKD7g7G5Mwu. 

And from the unit test of this code, there is no CIDR in this format, and the authors at that time should not support this format. https://github.com/istio/istio/blob/f371fe954459ef484fa2339afb37fce523b1bd52/pilot/pkg/networking/util/util_test.go#L84

**Comment by saltbo** [comment]
https://github.com/istio/istio/pull/40776  

@ramaraochavali  I found a fixed pr for the other CIDR test, maybe we should fix it too?

File: pilot/pkg/networking/util/util.go [hunk 1]
```diff
@@ -137,24 +128,48 @@ func ListContains(haystack []string, needle string) bool {
...
 	if strings.Contains(addr, "/") {
-		parts := strings.Split(addr, "/")
-		cidr.AddressPrefix = parts[0]
-		prefix, _ := strconv.Atoi(parts[1])
-		cidr.PrefixLen.Value = uint32(prefix)
+		ipp, err := netip.ParsePrefix(addr)
+		if err != nil {
+			return nil, err
+		}
+		ipAddr = ipp.Addr()
```
Comment (zhlsunshine) [inline pilot/pkg/networking/util/util.go:155]: I think `IsValid()` should be called before this specification.

Comment (saltbo) [reply pilot/pkg/networking/util/util.go:155]: I think `err != nil` already over the `IsValid()`

**Review by ramaraochavali — APPROVED** [review]

**Review by ericvn — APPROVED** [review]
Approve from a T&R standpoint

File: pilot/pkg/networking/util/util.go [hunk 2]
```diff
@@ -137,24 +128,48 @@ func ListContains(haystack []string, needle string) bool {
...
-	cidr := &core.CidrRange{
-		AddressPrefix: addr,
-		PrefixLen: &wrapperspb.UInt32Value{
-			Value: getMaxCidrPrefix(addr),
-		},
+	return cidr
+}
+
+// AddrStrToCidrRange converts from string to CIDR proto
+func AddrStrToCidrRange(addr string) (*core.CidrRange, error) {
```
Comment (aryan16) [inline pilot/pkg/networking/util/util.go:140]: since there were unit tests for  this method previously, can you add few test cases around this ?

Comment (saltbo) [reply pilot/pkg/networking/util/util.go:140]: revered the previous test cases

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**